### PR TITLE
HtmlRenderer: Also include whitespace when it is directly followed by an HTML tag

### LINKED ISSUE #512: Appropriate line breaks are not inserted for certain articles
**Newsboat version (copy the output of `newsboat -v` or the first line of `git show`)**:
r2.15-78-g37ca

Steps to reproduce the issue:

The following article shows the error https://blog.debiania.in.ua/posts/2019-05-05-i-underestimated-the-dangers-of-rust-ffi.html

Instead of proper line breaks you see

@Minoru thinks `newsboat` is not handling nested spans in the HTML properly.

### PR DESCRIPTION
**Pull request by dennisschagt**
Fixes #512

### CODE REVIEW

**Comment by dennisschagt** [comment]
### Rendered output demo

Old:
image

New:
image

File: src/tagsouppullparser.cpp [hunk 1]
```diff
@@ -75,7 +75,7 @@ TagSoupPullParser::Event TagSoupPullParser::next()
 	case Event::START_DOCUMENT:
 	case Event::START_TAG:
 	case Event::END_TAG:
-		skip_whitespace();
+		inputstream->read(&c, 1);
```
Comment (Minoru) [inline src/tagsouppullparser.cpp]: I think this change makes bda17e468e17e3e993c854a6a1af282673136a2b unnecessary, and it also means that 72a70b4774317f1928f6d82ebbd7383b63c0dc64 could be squashed into this commit.

File: src/tagsouppullparser.cpp [hunk 2]
```diff
@@ -606,9 +606,6 @@ void TagSoupPullParser::handle_tag()
 
 void TagSoupPullParser::handle_text()
 {
-	if (current_event != Event::START_DOCUMENT) {
```
Comment (Minoru) [inline src/tagsouppullparser.cpp]: I'd amend the previous commit, 1a9efe0de421d157eaf3f0755eb64126cdff37f5, instead of making a new one.

**Comment by dennisschagt** [comment]
>  I just wish you rewrote history a bit to remove the earlier attempt, making the changes easier to follow.

Just did that now.
I'm just a bit confused now by Github as it doesn't show the commits in the correct order (it shows them sorted by commit time instead of ordered as with `git log --reverse`).

> However, reading the changes one commit at a time felt a bit like going in circles. 

I will try to avoid that in future pull requests.

File: src/tagsouppullparser.cpp (hunk 1 above)
Comment (dennisschagt) [reply src/tagsouppullparser.cpp]: Done.
I've force-pushed after an interactive rebase.

To make sure the diff can still be seen, I've temporarily added a branch which points to the commit before the force push.
This shows that the end result is identical (just the commits have been reordered/combined): https://github.com/newsboat/newsboat/compare/234e26ac3a317e64bea23766ea0462f6a645772d..404c0896327224e5203f77c339984fe8be9b32e0

**Comment by Minoru** [comment]
> I'm just a bit confused now by Github as it doesn't show the commits in the correct order (it shows them sorted by commit time instead of ordered as with git log --reverse).

Wow, it does indeed! Now that's a funny choice.

Anyway, thanks for the work, lemme merge this!

**Pull request merged**